app/src/main/java/org/wikipedia/page/Namespace.kt

```diff
@@ -190,7 +193,7 @@ enum class Namespace(private val code: Int) : EnumCode {
             return if (UserTalkAliasData.valueFor(wiki.languageCode).equals(name, true) ||
                     UserTalkAliasData.valueFor(AppLanguageLookUpTable.FALLBACK_LANGUAGE_CODE).equals(name, true)) {
                 USER_TALK
-            } else MAIN
+            } else entries.firstOrNull { it.name.equals(name, true) } ?: MAIN
```

Comment:
This will only work for English. All namespaces have different localized names on each language wiki.

I think a better long-term solution will be to get rid of `fromLegacyString()` entirely, and have individual functions for `isTalk()`, `isUser()`, etc. and eliminate all the places where we attempt to check for `namespace() == MAIN`

Reply:
...and then we could also remove >100 of the actual Namespace enum definitions that we don't actually use anywhere, too.